### ISSUE TITLE
Possible to use int8 during pretraining stage?

### ISSUE THREAD

**Issue by 152334H** [haotian-liu/LLaVA#25]
Since the LLaMA model is frozen during pretraining, I was wondering if it was possible to run the model with int8 to reduce the vram used during training.

In my attempts to do so, I successfully started training with llama as int8 (by just adding the usual `load_in_8bit=True, device_map=...` combo), but observed that the loss would always collapse to `0.0` immediately. I also noticed the same issue when I simply loaded the llama model as fp16, rather than as bf16.

**Comment by 152334H** [comment]
also just an fyi but the references to `fastchat/train/train_mem.py` in the README should probably be changed to `llava/train/train_mem.py`

**Comment by haotian-liu** [comment]
Hi @152334H, thanks for your interest in our work!  Thanks for pointing out the script path issue, and I have just corrected it.

Regarding using INT8 during the pretraining stage, there _may_ be issues doing so.  The reason is that the Linear layer that is being projected, are optimized with the gradient that is backward propagated all the way from the last layer to the first layer of the LLM.  When directly doing this INT8 loading, I guess some gradients will underflow / overflow.

I do know that some Alpaca-LORA projects use INT8 training, and I am interested in integrating these techniques into our project as well.  I will also look into this after finishing all related release stuff.  Please let me know if you are also interested in this and would like to contribute!

**Comment by haotian-liu** [comment]
Hi, I am closing this issue, due to the inactivity.  Hope your problem has been resolved.  If you have further concerns, please feel free to re-open or open a new issue.  Thanks!

**Comment by 152334H** [comment]
No problem.
